### ISSUE TITLE
Block "Promoted Tweets" on mobile.twitter.com

### ISSUE THREAD

**Issue by spinda** [uBlockOrigin/uAssets#351]
### URL(s) where the issue occurs

`https://mobile.twitter.com` (login required?)

### Describe the issue

"Promoted Tweets" (ads) show up in the timeline feed for Twitter's mobile site.

### Versions

- Browser/version: Firefox Developer Edition (Aurora) 54.0a2 (2017-04-03) (64-bit)
- uBlock Origin version: uBlock Origin 1.11.4

### Notes

Adding the following custom filter rule blocks the promoted tweets:

```
mobile.twitter.com##div:has(:scope > div[role="article"] > div[data-testid="tweet"] > div > div:last-child:nth-child(3):nth-of-type(3))
```

`mobile.twitter.com` uses React and CSS Modules which obscures class names, so we search for tweets with three sections (header, body, "promoted" footer) instead, which only matches ads.

**Comment by gorhill** [comment]
I added a filter this morning for `mobile.twitter.com`, force an update of "uBlock filters" and see if this works.

**Comment by gorhill** [comment]
> uses React and CSS Modules which obscures class names

The class names look random, but are they really random or just meaningless? I was wondering when I created the filter.

**Comment by Jud** [comment]
I use this cosmetic filter, ymmv:

`mobile.twitter.com##:xpath(/html/body//div[@role="article"][.//text()[starts-with(., "Promoted")]])`

**Comment by gorhill** [comment]
This was fixed with b589a8e6.
